command/aarelayer/service/aa_types.go

```diff
@@ -0,0 +1,168 @@
+package service
+
+import (
+	"crypto/ecdsa"
+
+	"github.com/0xPolygon/polygon-edge/crypto"
+	"github.com/0xPolygon/polygon-edge/helper/keccak"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/fastrlp"
+)
+
+var marshalArenaPool fastrlp.ArenaPool
+
+// string mapping
+const (
+	StatusPending   string = "pending"   // The AA transaction is on the Pool
+	StatusQueued           = "queued"    // The AA transaction is waiting to be mined.
+	StatusCompleted        = "completed" // The `AA transaction` was mined in a block.
+	StatusFinished         = "finished"  // AA transaction` failed during the process.
```

Comment:
finished = failed?

Reply:
Just copied naming from rfc https://www.notion.so/polygontechnology/RFC-194-e14733e47dc548b4b12384965367ce3f. Maybe we should change there also?

Reply:
...but yes, it is semantically more correct if we change `StatusFinished` to `StatusFailed`, in my opinion you are right

Reply:
Changed https://github.com/0xPolygon/polygon-edge/pull/1252/commits/396931ce5428a8d78bb2bba0bfa8da056edc7fcd